sdk/core/core-client-rest/src/urlHelpers.ts

```diff
@@ -33,44 +76,97 @@ export function buildRequestUrl(
   );
 }
 
-function appendQueryParams(url: string, options: RequestParameters = {}) {
+function getQueryParamValue(
+  key: string,
+  allowReserved: boolean,
+  style: QueryParameterStyle,
+  param: any,
+): string {
+  let separator: string;
+  if (style === "pipeDelimited") {
+    separator = "|";
+  } else if (style === "spaceDelimited") {
+    separator = "%20";
+  } else {
+    separator = ",";
+  }
+
+  let paramValues: any[];
+  if (Array.isArray(param)) {
+    paramValues = param;
+  } else if (typeof param === "object" && param.toString === Object.prototype.toString) {
+    // If the parameter is an object without a custom toString implementation (e.g. a Date),
+    // then we should deconstruct the object into an array [key1, value1, key2, value2, ...].
+    paramValues = Object.entries(param).flat();
+  } else {
+    paramValues = [param];
+  }
+
+  const value = paramValues
+    .map((p) => {
+      if (p === null || p === undefined) {
+        return "";
+      }
+
+      if (!p.toString || typeof p.toString !== "function") {
+        throw new Error(`Query parameters must be able to be represented as string, ${key} can't`);
+      }
+
+      const rawValue = p.toISOString !== undefined ? p.toISOString() : p.toString();
```

Comment:
the premise here is we will serialize date type with toISOString by default ? however, if it's rfc7321 format or header date default type, we need  to use toUTCString instead. See typespec utcDateTime and offsetDateTime description here? https://typespec.io/docs/libraries/http/encoding#utcdatetime-and-offsetdatetime. I wonder if core could get such kind of format information?

Reply:
We also generate unixTimestamp as Date type as use getTime() to do the serialization.

Reply:
We are calling `toISOString` here in line with the existing behavior to avoid a breaking change. If we want to revisit this, we can, but I don't think this blocks this PR.

My understanding is that for dates in RLC we generate `string` if `toISOString` is unsuitable like in the cases you mentioned (@MaryGao may have more context).